homework2/2.1/Stack.Test/StackTest.cs

```diff
@@ -0,0 +1,53 @@
+using NUnit.Framework;
+using _2._1;
+using System.Collections.Generic;
+
+namespace StackTest
+{
+    public class StackTest
+    {
+        private static IEnumerable<TestCaseData> Stacks
+        => new TestCaseData[]
+        {
+             new TestCaseData(new StackArray()),
+             new TestCaseData(new StackList())
+        };
+
+        [Test, TestCaseSource(nameof(Stacks))]
+        public void PushThenStackIsNotEmpty(IStack stack)
+        {
+            var result = stack.IsEmpty();
+            stack.Push(1);
+            result = result && !stack.IsEmpty();
+            Assert.IsTrue(result);
```

Comment:
```suggestion
            Assert.IsTrue(stack.IsEmpty());
            stack.Push(1);
            Assert.IsFalse(stack.IsEmpty());
```
Никакие result-ы уже не нужны, можно делать проверки прямо по ходу теста. Если Assert не пройдёт, то тест прерывается, потому что нет смысла исполнять его дальше.